service/sdkbootstate.go

```diff
@@ -33,10 +36,41 @@ func (s *SDKBootState) Ready(rond sdk.OASEvaluatorFinder) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 	s.rond = rond
+	s.isReady = true
+	if s.ch != nil {
+		s.ch <- true
+		close(s.ch)
+		s.ch = nil
+	}
 }
 
 func (s *SDKBootState) Get() sdk.OASEvaluatorFinder {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 	return s.rond
 }
+
+func (s *SDKBootState) IsReady() bool {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	return s.isReady
+}
+
+func (s *SDKBootState) IsReadyChan() chan bool {
```

Comment:
bruh, this whole implementation is only here for testing...

Now I understand why you told me I'd not like this PR 🤣

Reply:
True 😄
But there is a drop of 10 seconds in the run of the tests (which on 40s average is not bad) due to the removal of the time.Sleep.
Otherwise, it could be possible to make a loop on the change of the isReady value: it's not cool, but it should work.

Reply:
![what-can-i-say-saturday-night-live](https://github.com/user-attachments/assets/b9ce3697-8aab-4253-a9ae-67adae7cf3a7)